engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -133,13 +133,13 @@ public boolean incrementEndpointServerErrors(UUID endpointId, int maxServerError
             if (endpoint.get().getServerErrors() + 1 > maxServerErrors) {
                 /*
                  * The endpoint exceeded the max server errors allowed from configuration.
-                 * It is therefore disabled and the errors counter is reset.
+                 * It is therefore disabled.
                  */
-                String hql = "UPDATE Endpoint SET serverErrors = 0, enabled = FALSE WHERE id = :id";
-                statelessSessionFactory.getCurrentSession().createQuery(hql)
+                String hql = "UPDATE Endpoint SET enabled = FALSE WHERE id = :id AND enabled IS TRUE";
+                int updated = statelessSessionFactory.getCurrentSession().createQuery(hql)
                         .setParameter("id", endpointId)
                         .executeUpdate();
-                return true;
+                return updated > 0;
```

Comment:
Now the endpoint can't be disabled twice even if there are 11, 12 or 13 server errors.

The returned boolean determines whether we should send an email notification or not.